alf/trainers/evaluator.py

```diff
@@ -74,10 +74,15 @@ def __init__(self, config: TrainerConfig, conf_file: str):
                       pre_configs, num_envs, config.root_dir, seed))
             self._worker.start()
         else:
-            self._env = create_environment(
-                for_evaluation=True,
-                num_parallel_environments=num_envs,
-                seed=seed)
+            if config.shared_train_eval_env:
+                assert not self._async, "should not use async_eval in shared_train_eval_env mode"
+                self._env = alf.get_env()
+                self._env.reset()
```

Comment:
Curious, why do we call env.reset() here but not in the other branch.  Maybe add a comment in the code?